java/src/jmri/jmrit/logixng/actions/swing/ActionListenOnBeansTableSwing.java

```diff
@@ -40,40 +40,18 @@ protected void createPanel(@CheckForNull Base object, @Nonnull JPanel buttonPane
 
         JPanel tableBeanPanelPanel = new JPanel();
         tableBeanPanelPanel.add(new JLabel(Bundle.getMessage("ActionListenOnBeansTableSwing_Table")));
-        tableBeanPanel = new BeanSelectPanel<>(InstanceManager.getDefault(NamedTableManager.class), null);
-        tableBeanPanelPanel.add(tableBeanPanel);
+        _tableBeanPanel = new BeanSelectPanel<>(InstanceManager.getDefault(NamedTableManager.class), null);
+        tableBeanPanelPanel.add(_tableBeanPanel);
         panel.add(tableBeanPanelPanel);
 
+        _rowOrColumnNameComboBox = new JComboBox<>();
         _tableRowOrColumnComboBox = new JComboBox<>();
         for (TableRowOrColumn item : TableRowOrColumn.values()) {
             _tableRowOrColumnComboBox.addItem(item);
         }
         JComboBoxUtil.setupComboBoxMaxRows(_tableRowOrColumnComboBox);
         _tableRowOrColumnComboBox.addActionListener((evt) -> {
-            _rowOrColumnNameComboBox.removeAllItems();
-            NamedTable table = tableBeanPanel.getNamedBean();
-            if (table != null) {
-                if (_tableRowOrColumnComboBox.getItemAt(_tableRowOrColumnComboBox.getSelectedIndex()) == TableRowOrColumn.Column) {
-                    for (int column=0; column <= table.numColumns(); column++) {
-                        // If the header is null or empty, treat the row as a comment
-                        Object header = table.getCell(0, column);
-                        if ((header != null) && (!header.toString().isEmpty())) {
-                            _rowOrColumnNameComboBox.addItem(header.toString());
-                        }
-                    }
-                } else {
-                    for (int row=0; row <= table.numRows(); row++) {
-                        // If the header is null or empty, treat the row as a comment
-                        Object header = table.getCell(row, 0);
-                        if ((header != null) && (!header.toString().isEmpty())) {
-                            _rowOrColumnNameComboBox.addItem(header.toString());
-                        }
-                    }
-                }
-                if (action != null) {
-                    _rowOrColumnNameComboBox.setSelectedItem(action.getRowOrColumnName());
-                }
-            }
+            comboListener(action);
         });
 
```

Comment:
Can you add these three lines after row 55?

```
        _tableBeanPanel.getBeanCombo().addActionListener((evt) -> {
            comboListener(action);
        });
```

So that the code looks like this:
```
        JComboBoxUtil.setupComboBoxMaxRows(_tableRowOrColumnComboBox);
        _tableRowOrColumnComboBox.addActionListener((evt) -> {
            comboListener(action);
        });
        _tableBeanPanel.getBeanCombo().addActionListener((evt) -> {
            comboListener(action);
        });
```

Reply:
No probs, have updated.